lib/timers/events.rb

```diff
@@ -60,18 +60,12 @@ def schedule(time, callback)
       # Maintain sorted order, O(logN) insertion time.
       @sequence.insert(index, handle)
       
-      return handle
+      handle
     end
     
     # Returns the first non-cancelled handle.
     def first
-      while handle = @sequence.last
-        if handle.cancelled?
-          @sequence.pop
-        else
-          return handle
-        end
-      end
+      @sequence.reverse.find { |handle| !handle.cancelled? }
```

Comment:
And the cancelled handles would only be popped up to the first uncancelled one. Seems like cleaning `@sequence` truly lumping that into `first` could help, or is the `pop` overhead expected to include the `pop` as a more indirect way of cleaning?